### PR TITLE
Wrap Statement Pointers to prevent use after free race

### PR DESCRIPTION
**Pull request by pyckle**
If Prepared Statements are misused in multiple threads (as can happen with a buggy prepared statement pool implementation), an application segmentation fault can occur because of a use after free condition.

Two unit tests that previously reliably caused segmentation faults. One uses a prepared statement while closing the connection in another thread. The other unit test uses the prepared statement, closing the prepared statement itself in another thread.

This fixes https://github.com/xerial/sqlite-jdbc/issues/509

### CODE REVIEW

File: src/main/java/org/sqlite/core/CoreStatement.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ protected CoreStatement(SQLiteConnection c) {
         rs = new JDBC4ResultSet(this);
     }
 
-    public DB getDatbase() {
+    public DB getDatabase() {
```
Comment (TassSinclair) [inline src/main/java/org/sqlite/core/CoreStatement.java:40]: 😁  Nice.

**Comment by pyckle** [comment]
@gotson  - please consider this for inclusion before releasing 3.39.2 Quite a few people (including myself) have hit this bug, which results in a cryptic jvm crash #719 #707 #680 and #509

It's java only changes, so no need to recompile the binaries.

This PR includes a unit test which reproduces the seg fault when the java code changes are not applied.

**Comment by gotson** [comment]
That's on my list, along with a few others.

I want to get the automatic snapshot published to the Maven snapshot repo too, so people can give a try to the fixes and confirm that it works.

**Comment by gotson** [comment]
Thanks a lot @pyckle for this PR. I am working to get snapshots published, so people can confirm if it fixes the issues #719 #680 #509

**Pull request merged**